File: pkg/dart2cpp/sample/dart/hello_test_working_enhanced.dart
/// Dart 工作增强语法测试示例
///
/// 包含转换器完全支持的Dart语法特性
///

void main() {
  print('🔥 Dart 工作增强语法测试开始');

  // 1. 基本数据类型
  testDataTypes();

  // 2. 控制流增强
  testWorkingControlFlow();

  // 3. 函数特性
  testWorkingFunctions();

  // 4. 集合特性
  testWorkingCollections();

  // 5. 类特性
  testWorkingClasses();

  // 6. 泛型特性
  testWorkingGenerics();

  print('✅ Dart 工作增强语法测试完成');
}

/// 测试数据类型
void testDataTypes() {
  print('\n--- 数据类型 ---');

  // 基本类型
  int age = 25;
  double price = 99.99;
  String name = '张三';
  bool isStudent = true;

  print('基本类型测试完成');
}

/// 测试工作控制流
void testWorkingControlFlow() {
  print('\n--- 工作控制流 ---');

  // switch 语句
  int score = 85;
  switch (score) {
    case 100:
    case 90:
      print('优秀');
      break;
    case 80:
      print('良好');
      break;
    default:
      print('其他分数');
  }

  // for-in 循环
  List<String> fruits = ['苹果', '香蕉', '橙子'];
  print('水果列表:');
  for (String fruit in fruits) {
    print('  $fruit');
  }

  // 条件表达式
  int testAge = 20;
  String description = testAge >= 18 ? '成年人' : '未成年人';
  print('年龄描述: $description');
}

/// 测试工作函数特性
void testWorkingFunctions() {
  print('\n--- 工作函数特性 ---');

  print('函数特性测试完成');
}

/// 测试工作集合特性
void testWorkingCollections() {
  print('\n--- 工作集合特性 ---');

  // List 操作
  List<int> numbers = [1, 2, 3, 4, 5];
  print('列表长度: ${numbers.length}');

  // Map 操作
  Map<String, int> scores = {'张三': 95, '李四': 87};
  print('映射大小: ${scores.length}');
}

/// 测试工作类特性
void testWorkingClasses() {
  print('\n--- 工作类特性 ---');

  // 创建对象
  WorkingPerson person = WorkingPerson('张三', 25);
  print('创建WorkingPerson对象');

  // 调用方法
  person.introduce();
}

/// 测试工作泛型特性
void testWorkingGenerics() {
  print('\n--- 工作泛型特性 ---');

  print('泛型特性测试完成');
}

// ========================
// 类定义
// ========================

/// 工作人员类
class WorkingPerson {
  String name;
  int age;

  WorkingPerson(this.name, this.age);

  /// 自我介绍
  void introduce() {
    print('大家好，我是 $name，今年 $age 岁。');
  }

  @override
  String toString() {
    return 'WorkingPerson{name: $name, age: $age}';
  }
}

File: pkg/dart2cpp/sample/cpp_generated/hello_test_working_enhanced.cpp
#include "dart2cpp.h"

// 工具宏定义

// ============================================================================
// 类: WorkingPerson
// ============================================================================

class WorkingPerson {
public:
  String name;
  Int age;
  WorkingPerson(String name, Int age) : name(name), age(age) {
  }
  
  Nullable introduce() {
    dart_print(dart_concat(dart_string("大家好，我是 "), (this->name).toString(), dart_string("，今年 "), (this->age).toString(), dart_string(" 岁。")));
return Void;
  }
  
  String toString() {
    return dart_concat(dart_string("WorkingPerson{name: "), (this->name).toString(), dart_string(", age: "), (this->age).toString(), dart_string("}"));
  }
  
};

Nullable testDataTypes();
Nullable testWorkingControlFlow();
Nullable testWorkingFunctions();
Nullable testWorkingCollections();
Nullable testWorkingClasses();
Nullable testWorkingGenerics();
Nullable testDataTypes() {
  dart_print(dart_string("\n--- 数据类型 ---"));
auto age = dart_int(25);
auto price = dart_double(99.99);
auto name = dart_string("张三");
auto isStudent = dart_bool(true);
dart_print(dart_string("基本类型测试完成"));
return Void;
}

Nullable testWorkingControlFlow() {
  dart_print(dart_string("\n--- 工作控制流 ---"));
auto score = dart_int(85);
if (score == dart_int(100) || score == dart_int(90)) {
  dart_print(dart_string("优秀"));
} else if (score == dart_int(80)) {
  dart_print(dart_string("良好"));
}else {
  dart_print(dart_string("其他分数"));
}
auto fruits = dart_literal<String>(dart_string("苹果"), dart_string("香蕉"), dart_string("橙子"));
dart_print(dart_string("水果列表:"));
auto sync_for_iterator = fruits->iterator();
for (; sync_for_iterator->hasNext(); ) {
auto fruit = sync_for_iterator->next();
dart_print(dart_string("  ") + (fruit).toString());
}
auto testAge = dart_int(20);
auto description = testAge->operator_greater_equals(dart_int(18)) ? dart_string("成年人") : dart_string("未成年人");
dart_print(dart_string("年龄描述: ") + (description).toString());
return Void;
}

Nullable testWorkingFunctions() {
  dart_print(dart_string("\n--- 工作函数特性 ---"));
dart_print(dart_string("函数特性测试完成"));
return Void;
}

Nullable testWorkingCollections() {
  dart_print(dart_string("\n--- 工作集合特性 ---"));
auto numbers = dart_literal<Int>(dart_int(1), dart_int(2), dart_int(3), dart_int(4), dart_int(5));
dart_print(dart_string("列表长度: ") + (numbers->size()).toString());
auto scores = Map<String, Int>::createFromEntries({{dart_string("张三"), dart_int(95)}, {dart_string("李四"), dart_int(87)}});
dart_print(dart_string("映射大小: ") + (scores->size()).toString());
return Void;
}

Nullable testWorkingClasses() {
  dart_print(dart_string("\n--- 工作类特性 ---"));
auto person = ObjectPtr<WorkingPerson>(new WorkingPerson(dart_string("张三"), dart_int(25)));
dart_print(dart_string("创建WorkingPerson对象"));
person->introduce();
return Void;
}

Nullable testWorkingGenerics() {
  dart_print(dart_string("\n--- 工作泛型特性 ---"));
dart_print(dart_string("泛型特性测试完成"));
return Void;
}

// ============================================================================
// 主函数
// ============================================================================

int main() {
  try {
    dart_print(dart_string("🔥 Dart 工作增强语法测试开始"));
testDataTypes();
testWorkingControlFlow();
testWorkingFunctions();
testWorkingCollections();
testWorkingClasses();
testWorkingGenerics();
dart_print(dart_string("✅ Dart 工作增强语法测试完成"));
    return 0;
  } catch (const std::exception& e) {
    std::cerr << "Error: " << e.what() << std::endl;
    return 1;
  }
}
